update: Remove Wayland section from and add Permission Controls section to Linux Overview (#2690)
privacyguides/privacyguides.org: docs/os/linux-overview.md

Additionally, some distributions will not download firmware updates automatically. For that, you will need to install [`fwupd`](https://wiki.archlinux.org/title/Fwupd).

### Permission Controls

> Desktop environments (DEs) that support the [Wayland](https://wayland.freedesktop.org) display protocol are overall improvements [security-wise](https://lwn.net/Articles/589147) over those that support only X11. However, not all DEs implement security practices that take advantage of the architectural improvements that Wayland brings.

Reviewer: "overall improvements security-wise" - bit of a mouthful

```suggestion
Desktop environments (DEs) that support the [Wayland](https://wayland.freedesktop.org) display protocol are [more secure](https://lwn.net/Articles/589147) than those that only support X11. However, not all DEs take full advantage of Wayland's architectural security improvements.
```

> For example, GNOME has a notable edge in security compared to other DEs by implementing [permission controls](https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1970) for third-party software that tries to [capture your screen](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/3943). That is, when a third-party screenshot application like [Flameshot](https://flameshot.org) attempts to capture your screen for a screenshot, you are prompted for your permission to share your screen to the app.

Reviewer: - The merge request description at the first link is overly technical I think, I'm not sure what the value is.
- Why are we linking to this specific application?

```suggestion
For example, GNOME has a notable edge in security compared to other DEs by implementing permission controls for third-party software that tries to [capture your screen](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/3943). That is, when a third-party application attempts to capture your screen, you are prompted for your permission to share your screen with the app.
```

> <figure markdown>
>   ![Screenshot permissions](../assets/img/linux/screenshot_permission.png)
>   <figcaption>GNOME's screenshot permission dialog</figcaption>
> </figure>

Reviewer: ```suggestion
<figure markdown>
  ![Screenshot permissions](../assets/img/linux/screenshot_permission.png){ width="450" }
  <figcaption>GNOME's screenshot permission dialog</figcaption>
</figure>
```

> For other DEs and window managers like [KDE](https://kde.org) and [Sway](https://swaywm.org), permission controls for such privileged actions [do not yet exist](https://invent.kde.org/plasma/xdg-desktop-portal-kde/-/issues/7), or software maintainers are [waiting for the upstream](https://github.com/swaywm/sway/issues/5118#issuecomment-600054496) Wayland protocol to implement these controls. It is up to you to decide, based on your [threat model](../basics/threat-modeling.md), whether the absence of these permission controls is impactful in your choice of what desktop environment to use.

Reviewer: ```suggestion
Many alternatives don't provide these same permission controls yet,[^1] while some are waiting for Wayland protocol to implement these controls upstream.[^2]

[^1]: KDE currently has an open proposal to add controls for screen captures: <https://invent.kde.org/plasma/xdg-desktop-portal-kde/-/issues/7>
[^2]: Sway is waiting to add specific security controls until they "know how security as a whole is going to play out" in Wayland: <https://github.com/swaywm/sway/issues/5118#issuecomment-600054496>
```

- Regarding threat modeling, does this have to be stated? It feels redundant.
- Linking to product homepages in this context feels extraneous, we shouldn't direct readers to links that don't provide direct relevance to the sentence they're reading. Also note we didn't link to GNOME above either anyways.

Author: > Linking to product homepages in this context feels extraneous, we shouldn't direct readers to links that don't provide direct relevance to the sentence they're reading.

Yeah, that makes a lot of sense :+1: will keep this in mind moving forward

Additionally, some distributions will not download firmware updates automatically. For that, you will need to install [`fwupd`](https://wiki.archlinux.org/title/Fwupd).

### Permission Controls

Desktop environments (DEs) that support the [Wayland](https://wayland.freedesktop.org) display protocol are [more secure](https://lwn.net/Articles/589147) than those that only support X11. However, not all DEs take full advantage of Wayland's architectural security improvements.

> For example, GNOME has a notable edge in security compared to other DEs by implementing permission controls for third-party software that tries to [capture your screen](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/3943). That is, when a third-party application attempts to capture your screen, you are prompted for your permission to share your screen with the app.

Reviewer 2: JFYI this isn't quite accurate. We have this as well in KWin, for 3rd-party apps using the portal API. The only place it's not used is for Spectacle, which is our own 1st party screenshot app. Spectacle bypasses the portal API because its UX isn't compatible with the UX assumptions made by the designers of the portal API. This is the part that's an open topic of discussion.

Reviewer 2: Upon further review of our code, it seems that you were right and I was wrong. This does indeed seem to be something we should improve. Sorry for the noise.
